VideoLocker/src/main/java/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -135,79 +136,63 @@ public void onClick(View v) {
+                        if (fieldView != null) mFieldViews.add(fieldView);
+                    }
+                }
 
-            for (RegistrationFormField field : form.getFields()) {
-                if (field.getFieldType().equals(RegistrationFieldType.CHECKBOX)
-                        && field.getAgreement() != null) {
-                    // this is agreement field
-                    // this must be added at the end of the form
-                    // hold on it
-                    agreements.add(field);
-                } else {
-                    IRegistrationFieldView fieldView = IRegistrationFieldView.Factory.getInstance(inflater, field);
-                    if (fieldView != null) mFieldViews.add(fieldView);
+                // add required and optional fields to the window
+                for (IRegistrationFieldView v : mFieldViews) {
+                    if (v.getField().isRequired()) {
+                        requiredFieldsLayout.addView(v.getView());
+                    } else {
+                        optionalFieldsLayout.addView(v.getView());
+                    }
                 }
-            }
 
-            // add required and optional fields to the window
-            for (IRegistrationFieldView v : mFieldViews) {
-                if (v.getField().isRequired()) {
-                    requiredFieldsLayout.addView(v.getView());
-                } else {
-                    optionalFieldsLayout.addView(v.getView());
+                // add agreement fields to the window if available
+                for (RegistrationFormField agreement : agreements) {
+                    IRegistrationFieldView agreementView = IRegistrationFieldView.Factory.getInstance(inflater, agreement);
+                    agreementView.setActionListener(new IRegistrationFieldView.IActionListener() {
+                        @Override
+                        public void onClickAgreement() {
+                            showAgreement();
+                        }
+                    });
+                    agreementLayout.addView(agreementView.getView());
                 }
-            }
 
-            // add agreement fields to the window if available
-            for (RegistrationFormField agreement : agreements) {
-                IRegistrationFieldView agreementView = IRegistrationFieldView.Factory.getInstance(inflater, agreement);
-                agreementView.setActionListener(new IRegistrationFieldView.IActionListener() {
-                    @Override
-                    public void onClickAgreement(RegistrationAgreement agreement) {
-                        showAgreement(agreement);
-                    }
-                });
-                agreementLayout.addView(agreementView.getView());
+                // request rendering of the layouts
+                requiredFieldsLayout.requestLayout();
+                optionalFieldsLayout.requestLayout();
+                agreementLayout.requestLayout();
```

Comment:
nit: `requestLayout()` is already called by `addView`. I think you can delete these 3 lines.